src/main/java/net/pms/util/MimeType.java

```diff
@@ -0,0 +1,327 @@
+					return o1.compareToIgnoreCase(o2);
+				}
+
+			});
+			for (Entry<String, String> entry : parameters.entrySet()) {
+				map.put(entry.getKey(), entry.getValue());
+			}
+			this.parameters = Collections.unmodifiableSortedMap(map);
+		}
+	}
+
+	/**
+	 * @return The {@code type} (first) part of this {@link MimeType}.
+	 */
+	public String getType() {
+		return type;
+	}
+
+	/**
+	 * @return Whether the {@code type} (first) part of this
+	 *         {@link MimeType} matches anything.
+	 */
+	public boolean isAnyType() {
+		return isBlank(type) || ANY.equals(type);
+	}
+
+	/**
+	 * @return The {@code subtype} (second) part of this {@link MimeType}.
+	 */
+	public String getSubtype() {
+		return subtype;
+	}
+
+	/**
+	 * @return Whether the {@code subtype} (second) part of this
+	 *         {@link MimeType} matches anything.
+	 */
+	public boolean isAnySubtype() {
+		return ANY.equals(subtype);
+	}
+
+	/**
+	 *
+	 * @return A {@link Map} containing the {@code parameters} of this
+	 *         {@link MimeType}. If no parameters exist, an empty
+	 *         {@link Map} is returned, not {@code null}.
+	 */
+	public Map<String, String> getParameters() {
+		return parameters;
+	}
+
+	/**
+	 * Determines whether this and {@code other} is compatible. That means being
+	 * either equal or having {@link #ANY} in {@code type} or {@code subtype} is
+	 * such a way that they can describe the same content without taking
+	 * parameters into account.
+	 *
+	 * @param other the {@link MimeType} to check compatibility against.
+	 * @return {@code true} of this and {@code other} is compatible,
+	 *         {@code false} otherwise.
+	 */
+	public boolean isCompatible(MimeType other) {
+		if (other == null) {
+			return false;
+		}
+		if (
+			(isBlank(type) || ANY.equals(type)) &&
+			isBlank(subtype) ||
+			(isBlank(other.type) || ANY.equals(other.type)) &&
+			isBlank(other.subtype)
+		) {
+			return true;
+		} else if (isBlank(type) || (isBlank(other.type))) {
+			return
+				isBlank(subtype) ||
+				isBlank(other.subtype) ||
+				ANY.equals(subtype) ||
+				ANY.equals(other.subtype) ||
+				subtype.toLowerCase(Locale.ROOT).equals(other.subtype.toLowerCase(Locale.ROOT));
+		} else if (
+			type.toLowerCase(Locale.ROOT).equals(other.type.toLowerCase(Locale.ROOT)) &&
+			(
+				isBlank(subtype) ||
+				ANY.equals(subtype) ||
+				isBlank(other.subtype) ||
+				ANY.equals(other.subtype)
+			)
+		) {
+			return true;
+		} else if (isBlank(subtype) || isBlank(other.subtype)) {
+			return false;
+		} else {
+			return
+				type.toLowerCase(Locale.ROOT).equals(other.type.toLowerCase(Locale.ROOT)) &&
+				subtype.toLowerCase(Locale.ROOT).equals(other.subtype.toLowerCase(Locale.ROOT));
+		}
+	}
+
+	/**
+	 * Creates a new {@link MimeType} by attempting to parse
+	 * {@code stringValue}.
+	 *
+	 * @param stringValue the {@link String} to parse.
+	 * @return The new {@link MimeType}.
+	 * @throws ParseException If {@code stringValue} isn't a valid mime-type.
```

Comment:
@Sami32 There's really no rules for capitalization as far as I know, I see both upper and lower case used in JavaDocs for the start of sentences. My personal preference is to use lower case for parameters and upper case for ```@return``` and ```@throws```. The reason for this is the way the JavaDoc is formatted, where the parameter description follows immediately after the parameter name - while the return and throws descriptions are more "separated" and thus I see them as separate sentences. So, no type, just my preference.